### DISCUSSION TITLE
How is `main` function in 'erlang_ls' called?

### DISCUSSION THREAD

**Discussion by mdbergmann** [erlang-ls/erlang_ls#1214] [General]
Hello.

I'm still a bit new to Erlang. So, sorry for this of topic question.
In my 'release' app I'd also want to allow to pass in command line parameters.
So I'm wondering how in erlang_ls.erl the `main` function is actually called, what is the wiring that is required for this to work?
I couldn't find anything in erlang_ls that I don't have, but then, it must be something.

Thanks,
Manfred

**Comment by michalmuskala** [comment]
Erlang LS uses the "escript" feature to generate the executable - this includes calling the `main` function as the entry-point to the program.

You can learn more here: https://www.erlang.org/doc/man/escript.html and here: https://rebar3.org/docs/commands/#escriptize

**Reply by mdbergmann** [reply]
Thanks a lot I got it working.
